services/src/main/java/org/keycloak/services/resources/admin/RealmsAdminResource.java

```diff
@@ -92,10 +93,13 @@ public RealmsAdminResource(AdminAuth auth, TokenManager tokenManager) {
     @NoCache
     @Produces(MediaType.APPLICATION_JSON)
     public List<RealmRepresentation> getRealms() {
+        Map<String, String> realmLocalization;
+        String userLocale = auth.getToken().getLocale();
         List<RealmRepresentation> reps = new ArrayList<RealmRepresentation>();
         List<RealmModel> realms = session.realms().getRealms();
         for (RealmModel realm : realms) {
-            addRealmRep(reps, realm);
+            realmLocalization = session.realms().getRealmLocalizationByLocale(realm.getId(), userLocale);
```

Comment:
I would avoid having the localization from the basic realm representation but use the new endpoint to obtain them, if necessary.